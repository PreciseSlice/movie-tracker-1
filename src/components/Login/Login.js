import React, { Component } from 'react';
import { connect } from 'react-redux';
// import PropTypes from 'prop-types';

import { fetchUser } from '../../helpers/helper';
import { getUser, login } from '../../actions/index';
import PropTypes from 'prop-types';

class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: ''
    };
  }

  async validateLogin() {
    const { handleLogin } = this.props;
    const { email, password } = this.state;
    const userFetchResponse = await fetchUser();

    const userMatch = userFetchResponse.data.find(user => {
      return user.email === email && user.password === password;
    });
    if (userMatch) {
      handleLogin(true);
    } else {
      throw new Error('Email address and password not found!');

    }
  }

  render() {
    const { handleSubmit } = this.props;
    return (
      <section>
        <form
          onSubmit={event => {
            event.preventDefault();
            handleSubmit(this.state.email.toLowerCase(), this.state.password);
            this.validateLogin();
          }}
        >
          <input
            value={this.state.email}
            placeholder="email"
            type="email"
            autoComplete="email"
            onChange={event => this.setState({ email: event.target.value })}
            required
          />
          <input
            value={this.state.password}
            placeholder="password"
            type="password"
            autoComplete="on"
            onChange={event => this.setState({ password: event.target.value })}
            required
          />
          <button>Log In</button>
        </form>
      </section>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  handleSubmit: (email, password) => dispatch(getUser(email, password)),
  handleLogin: boolean => dispatch(login(boolean))
});

Login.propTypes = {
  handleSubmit: PropTypes.func
};

export default connect(null, mapDispatchToProps)(Login);
