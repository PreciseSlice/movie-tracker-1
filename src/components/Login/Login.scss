form {
  display: flex;
  align-items: center;
  flex-direction: column;
  margin: 20px 0 0 0;
}

input {
  margin: 7px 0;
  width: 40%;
  height: 30px;
  text-indent: 7px;
  letter-spacing: .7px;
  font-size: 1em;
}

form button {
  border: none;
  border-radius: 3px;
  background-color: #ceb3c5;
  color: #fff;
  font-size: 1.3em;
  height: 36px;
  letter-spacing: 1.2px;
  width: 40.5%;
  &:hover {
    background-color: #9c7e92
  }
}