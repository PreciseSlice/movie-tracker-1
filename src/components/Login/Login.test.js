/* eslint-disable */
import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Login from './Login';

//Enzyme.configure({ adapter: new Adapter() });

describe('Login', () => {
    it('matches the snapshot', () => {
        // const wrapper = shallow(<Login />);
        
        // expect(wrapper).toMatchSnapshot();
    });
});