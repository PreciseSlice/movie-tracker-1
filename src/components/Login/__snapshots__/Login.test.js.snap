// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches the snapshot 1`] = `
<section>
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="email"
      onChange={[Function]}
      placeholder="email"
      required={true}
      type="email"
      value=""
    />
    <input
      autoComplete="on"
      onChange={[Function]}
      placeholder="password"
      required={true}
      type="password"
      value=""
    />
    <button>
      Log In
    </button>
  </form>
</section>
`;
